.circleci/config.yml

```diff
@@ -19,6 +19,11 @@ jobs:
       JVM_OPTS: -Xmx3200m
     steps:
       - checkout
+
+      - run:
+          name: Check Code Format
```

Comment:
The CircleCI configuration contains two jobs which run in parallel, one for building and running the Java unit tests and one for running the Android functional tests. I've only added this Check Code Format step to the `build-and-unit-test` job. If it fails it will fail the entire build, but the `functional-test` just will run to completion. Should I additionally add the Check Code Format step to the `functional-test` job so both jobs fail?

Reply:
Ideally we should run the build and all the quick checks first, so that you don't even run the tests if the build job didn't pass. Can we create a new job for the build and reference it with `requires` in the test jobs?

Reply:
Okay, added that. It only added about 35 seconds to the build time.